app/upgrades/v11/upgrades.go

```diff
@@ -9,13 +9,24 @@ import (
 	"github.com/osmosis-labs/osmosis/v7/app/upgrades"
 )
 
+// We set the app version to pre-upgrade because it will be incremented by one
+// after the upgrade is applied by the handler.
+const preUpgradeAppVersion = 10
+
 func CreateUpgradeHandler(
 	mm *module.Manager,
 	configurator module.Configurator,
 	bpm upgrades.BaseAppParamManager,
 	keepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// Although the app version was already set during the v9 upgrade, our v10 was a fork.
+		// As a result, the upgrade handler was not executed to increment the app version.
+		// This change helps to correctly set the app version for v11.
+		if err := keepers.UpgradeKeeper.SetAppVersion(ctx, preUpgradeAppVersion); err != nil {
+			return nil, err
+		}
```

Comment:
Do we have to do this manually every upgrade? Or is this just one-off code that we only have to do once?

Reply:
> Do we have to do this manually every upgrade?

No, we need to do this now because the last release was a fork.

> Also why does the API take in pre upgrade version, not post?

This is because this logic is run before the upgrade handler logic that increases this version by 1